lib/order.rb

```diff
@@ -0,0 +1,79 @@
+require_relative 'customer'
+require 'csv'
+# require 'awesome_print'
+
+class Order
+  attr_reader :id, :products, :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+
+    if !([:pending, :paid, :processing, :shipped, :complete].include?(@fulfillment_status))
+      raise ArgumentError.new("Unknown fulfillment status")
+    end
+
+  end
+
+  def total
+    @total = (@products.values.sum * 1.075).round(2)
+  end
+
+  def add_product(product_name, price)
+    if @products.has_key?(product_name)
+      raise ArgumentError.new("Product is already in the order!")
+    else
+      @products[product_name] = price
+    end
+  end
+
+  def remove_product(product_name)
+    if @products.has_key?(product_name)
+      @products.delete(product_name)
+    else
+      raise ArgumentError.new("No product with that name was found!")
+    end
+  end
+
+  def self.products_to_hash(products_str)
```

Comment:
👍 , This is a great helper method.  I do suggest making it a private method.

It would also be good to use `.map` here.